src/utils/include/utils/mpi/gatherv.hpp

```diff
@@ -61,6 +61,7 @@ void gatherv_impl(const boost::mpi::communicator &comm, const T *in_values,
 
     /* not in-place */
     if (in_values != out_values) {
+      // NOLINTNEXTLINE(clang-analyzer-core.NullDereference)
```

Comment:
This can be easily fiixed: `if(in_value && out_value && in_values != out_values)`